services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_compose_specs.py

```diff
@@ -251,7 +253,7 @@ def assemble_spec(
         }
         container_name = DEFAULT_SINGLE_SERVICE_NAME
     else:
-        service_spec = compose_spec
+        service_spec = deepcopy(compose_spec)
```

Comment:
@sanderegg this is the core of the issue! 

The `compose_spec` field is stored in the `SchedulerData` object which keeps track of the state of the sidecar.

If the director-v2 fails to start the user services (aka: ask sidecar to run docker-compose up), the content of `compose_spec` on the next retry run is different.

This is an unwanted side effect which slipped by us since the beginning.

Reply:
Great! this seems more likely to be the issue. cool thanks for double checking!